docs/storage.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Storage
+sort_rank: 5
+---
+
+# Storage
+
+Prometheus includes a local on-disk time series database, but also optionally integrates with remote storage systems.
+
+## Local storage
+
+Prometheus's local time series database stores time series data in a custom format on disk.
+
+### On-disk layout
+
+Ingested samples are grouped into blocks of two hours (by default). Each two-hour block consists of a directory containing one or more chunk files that contain all time series samples for that window of time, as well as a metadata file and index file (which indexes metric names and labels to series in the chunk files). The block for currently incoming samples is kept in memory and not fully persisted yet. It is secured against crashes by a write-ahead-log (WAL) that can be replayed when the Prometheus server restarts after a crash. When series are deleted via the API, deletion records are stored in separate tombstone files (instead of deleting the data immediately from the chunk files).
+
+The directory structure of a Prometheus server's data directory will look something like this:
+
+```
+./data/01BKGV7JBM69T2G1BGBGM6KB12
+./data/01BKGV7JBM69T2G1BGBGM6KB12/meta.json
+./data/01BKGV7JBM69T2G1BGBGM6KB12/wal
+./data/01BKGV7JBM69T2G1BGBGM6KB12/wal/000002
+./data/01BKGV7JBM69T2G1BGBGM6KB12/wal/000001
+./data/01BKGTZQ1SYQJTR4PB43C8PD98
+./data/01BKGTZQ1SYQJTR4PB43C8PD98/meta.json
+./data/01BKGTZQ1SYQJTR4PB43C8PD98/index
+./data/01BKGTZQ1SYQJTR4PB43C8PD98/chunks
+./data/01BKGTZQ1SYQJTR4PB43C8PD98/chunks/000001
+./data/01BKGTZQ1SYQJTR4PB43C8PD98/tombstones
+./data/01BKGTZQ1HHWHV8FBJXW1Y3W0K
+./data/01BKGTZQ1HHWHV8FBJXW1Y3W0K/meta.json
+./data/01BKGTZQ1HHWHV8FBJXW1Y3W0K/wal
+./data/01BKGTZQ1HHWHV8FBJXW1Y3W0K/wal/000001
+./data/01BKGV7JC0RY8A6MACW02A2PJD
+./data/01BKGV7JC0RY8A6MACW02A2PJD/meta.json
+./data/01BKGV7JC0RY8A6MACW02A2PJD/index
+./data/01BKGV7JC0RY8A6MACW02A2PJD/chunks
+./data/01BKGV7JC0RY8A6MACW02A2PJD/chunks/000001
+./data/01BKGV7JC0RY8A6MACW02A2PJD/tombstones
+```
+
+The initial two-hour blocks are eventually compacted into longer blocks in the background.
+
+Note that a limitation of the local storage is that it is not clustered or replicated. Thus, it is not arbitrarily scalable or durable in the face of disk or node outages and should thus be treated as more of an ephemeral sliding window of recent data. However, if your durability requirements are not strict, you may still succeed in storing up to years of data in the local storage.
```

Comment:
We usually use the word cache

Reply:
"Cache" is taking the wording a bit too far IMO, as it carries a connotation of even less guarantees of persistence than we have. Basically, that Prometheus may actively blow away random parts of the storage at any time.